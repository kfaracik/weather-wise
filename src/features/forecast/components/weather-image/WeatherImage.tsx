import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import clearSkyImage from '@shared/assets/images/sun.webp';
import partlyCloudyImage from '@shared/assets/images/partlycloudy.webp';
import moderateRainImage from '@shared/assets/images/moderaterain.webp';
import mistImage from '@shared/assets/images/mist.webp';
import cloudyImage from '@shared/assets/images/cloud.webp';
import heavyRainImage from '@shared/assets/images/heavyrain.webp';

const getWeatherImage = (weatherMain: string, weatherDescription: string) => {
  if (weatherMain === 'Clear') {
    return clearSkyImage; 
  } else if (weatherMain === 'Clouds') {
    if (weatherDescription.includes('partly cloudy')) {
      return partlyCloudyImage;
    } else if (weatherDescription.includes('cloudy')) {
      return cloudyImage;
    }
    return partlyCloudyImage;
  } else if (weatherMain === 'Rain') {
    if (weatherDescription.includes('heavy')) {
      return heavyRainImage;
    } else if (weatherDescription.includes('moderate')) {
      return moderateRainImage;
    }
    return moderateRainImage;
  } else if (weatherMain === 'Mist' || weatherMain === 'Fog') {
    return mistImage;
  } else {
    return clearSkyImage;
  }
};

export const WeatherImage = ({
  weather,
}: {
  weather: { main: string; description: string }[];
}) => {
  const weatherCondition = weather[0];
  const weatherImage = getWeatherImage(
    weatherCondition.main,
    weatherCondition.description,
  );

  return (
    <View style={styles.container}>
      <Image source={weatherImage} style={styles.image} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: 100,
    height: 100,
    resizeMode: 'contain',
  },
});
