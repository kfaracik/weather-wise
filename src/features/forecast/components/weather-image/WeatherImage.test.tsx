import React from 'react';
import {render} from '@testing-library/react-native';
import {WeatherImage} from './WeatherImage';

describe('WeatherImage', () => {
  it('should generate snapshot', () => {
    const {toJSON} = render(
      <WeatherImage weather={[{main: 'Clouds', description: 'moderate'}]} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
