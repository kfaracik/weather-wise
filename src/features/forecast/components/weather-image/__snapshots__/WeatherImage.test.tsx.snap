// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherImage should generate snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/shared/assets/images/partlycloudy.webp",
      }
    }
    style={
      {
        "height": 100,
        "resizeMode": "contain",
        "width": 100,
      }
    }
  />
</View>
`;
